### PR TITLE
CP-33121: Stop using Xstringext in cli_operations

### PR DESCRIPTION
**Pull request by psafont**
Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

### CODE REVIEW

File: ocaml/xapi/cli_operations.ml [hunk 1]
```diff
@@ -167,7 +169,9 @@ let diagnostic_net_stats printer rpc session_id params =
   let uri (_, u, _) =
     not (List.mem_assoc "uri" params)
     || (String.lowercase_ascii u = String.lowercase_ascii (List.assoc "uri" params)) in
-  let has_param x = not(List.mem_assoc "params" params) || (List.mem x (String.split ',' (List.assoc "params" params))) in
+  let has_param x =
+    not (List.mem_assoc "params" params)
+    || (List.mem x (String.split_on_char ',' (List.assoc "params" params))) in
```
Comment (edwintorok) [inline ocaml/xapi/cli_operations.ml]: can this use List.assoc_opt here instead of mem+assoc?

File: ocaml/xapi/cli_operations.ml [hunk 2]
```diff
@@ -832,7 +839,10 @@ let pool_disable_binary_storage printer rpc session_id params =
 
 let pool_ha_enable printer rpc session_id params =
   let config = read_map_params "ha-config" params in
-  let uuids = if List.mem_assoc "heartbeat-sr-uuids" params then String.split ',' (List.assoc "heartbeat-sr-uuids" params) else [] in
+  let uuids = if List.mem_assoc "heartbeat-sr-uuids" params then
```
Comment (edwintorok) [inline ocaml/xapi/cli_operations.ml]: assoc_opt

File: ocaml/xapi/cli_operations.ml (hunk 1 above)
Comment (psafont) [reply ocaml/xapi/cli_operations.ml]: Can't, this `List` is from stdext, I'll add it to https://github.com/xapi-project/xen-api/pull/4038 I already have removed Listext for this reason.

Comment (psafont) [reply ocaml/xapi/cli_operations.ml]: See https://github.com/xapi-project/xen-api/commit/acd69b306

**Comment by lindig** [comment]
The code for handling parameter lists is a good candidate for simplification. I understand that this is not the point of this PR but I am horrified by the endless string matching and splitting. A first step would be to document what key and values we actually support and to come up with an interface for a sub module.

File: ocaml/xapi/cli_operations.ml [hunk 3]
```diff
@@ -1871,7 +1885,9 @@ let select_vms ?(include_control_vms = false) ?(include_template_vms = false) rp
   let params = if not include_template_vms then ("is-a-template"    , "false") :: params else params in
   let vm_name_or_ref = try Some (
       (* Escape every quote character *)
-      List.assoc "vm" params |> String.replace "\"" "\\\""
+      let quote = Re.(compile (str {|"|})) in
```
Comment (edwintorok) [inline ocaml/xapi/cli_operations.ml]: you could compile the regex just once if you put this somewhere global

Comment (psafont) [reply ocaml/xapi/cli_operations.ml]: I've moved it outside the function and reused it for sr and host parameters.

**Pull request merged**